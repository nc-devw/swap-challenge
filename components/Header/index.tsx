import React from "react";

import Link from "next/link";
import Image from "next/image";
import Logo from "../../public/assets/logo.png";

const Header = () => {
  return (
    <div className="py-4 px-12 shadow bg-white">
      <div className="flex text-white font-semibold items-center text-neutral-900">
        <div className="flex-[1] flex">
          <div className="w-6 h-6 rounded-full overflow-hidden">
            <Image
              src={Logo}
              width="100%"
              height="100%"
              layout="responsive"
              objectFit="contain"
            />
          </div>
          <span className="ml-2">Belo Challenge</span>
        </div>
        <div className="flex-[4] flex gap-5">
          <Link href="/">Home</Link>
          <Link href="/swap">Swap</Link>
          <Link href="/swap-status">Swap Status</Link>
        </div>
      </div>
    </div>
  );
};

export default Header;
