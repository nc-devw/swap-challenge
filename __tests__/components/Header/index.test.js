import { render } from "@testing-library/react";
import Header from "@/components/Header";

describe("Header", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment).toMatchSnapshot();
  });
});
