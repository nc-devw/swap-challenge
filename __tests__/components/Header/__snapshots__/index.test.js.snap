// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<DocumentFragment>
  <div
    class="py-4 px-12 shadow bg-white"
  >
    <div
      class="flex text-white font-semibold items-center text-neutral-900"
    >
      <div
        class="flex-[1] flex"
      >
        <div
          class="w-6 h-6 rounded-full overflow-hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
            />
            <img
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
            />
            <noscript />
          </span>
        </div>
        <span
          class="ml-2 hidden md:block"
        >
          Belo Challenge
        </span>
      </div>
      <div
        class="flex-[4] flex gap-5"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/swap"
        >
          Swap
        </a>
        <a
          href="/transactions"
        >
          Transactions
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
